### DISCUSSION TITLE
How to return a lualine component using a function?

### DISCUSSION THREAD

**Discussion by LoyalPotato** [nvim-lualine/lualine.nvim#1143] [Q&A]
Hello,
I'm using [nvim-navic](https://github.com/SmiteshP/nvim-navic) in the winbar to get some info according to the location of the cursor.
But when the cursor is in an empty line, that isn't inside a function or similar, it won't show anything. So what I'm trying to do is, when this is the case, show one of the components from lualine, such as `filename`.

The config for lualine's winbar is like so:
```lua
    winbar = {
        lualine_c = {
            {
                function()
                    if navic.is_available() then
                        local loc = navic.get_location()
                        if (loc == nil or loc == '') then
                            -- tried: return 'filename', return require'lualine.components.filename'()
                            return require'lualine.components.filename'
                        end
                        return loc
                    end
                end,
                draw_empty = true,
            }
        }
    },
```

Is this even possible to do in a straightforward way?

**Comment by acook** [comment]
I'm trying to do something similar by re-using Lualine's components in my `tmux` tabs, but Lualine is definitely doing some interesting things. There is a kind of class implementation around components, so filename descends from `component`. This adds some complexity. Plus, when a built-in component is required, it must be initialized before it can do anything. So the initial configuration must be preserved or extracted to get the same output.

I know it's possible to wrangle I ended up just writing my own function.
